src/richie/apps/search/signals.py

```diff
@@ -165,3 +169,19 @@ def on_page_unpublished(sender, instance, language, **kwargs):
         transaction.on_commit(
             lambda: apply_es_action_to_page(instance, action, language)
         )
+
+
+# pylint: disable=unused-argument
+@receiver(post_obj_operation)
+def on_page_moved(sender, **kwargs):
+    """
+    When a page is moved, we may need to re-index all pages linked to objects of
+    the same kind. This applies to all pages linked to categories as they have the
```

Comment:
I misunderstood what you meant by "pages linked to":
```suggestion
    the same kind. This applies to all category pages as they have the
```